### ISSUE TITLE
4.6 beta - Custom extension doesn't load

### ISSUE THREAD

**Issue by antonbondarchuk** [eirannejad/pyRevit#383]
Hi Ehsan,

Could you have a look into it..Maybe I'm doing something wrong, but after upgrading to 4.6-beta I'm having a hard time to load my custom extension (was not an issue with 4.5 and I haven't renamed-moved directory where the x.extension lives).. traceback pdf attached
Thanks, Anton
[pyRevitErr.pdf](https://github.com/eirannejad/pyRevit/files/2462444/pyRevitErr.pdf)

**Comment by eirannejad** [comment]
Hmm. What's the file path to your extension? Would you be able to look into your `%appdata%/pyRevit/pyRevit-config.ini` file and see what is set for `userextensions` value?

**Comment by antonbondarchuk** [comment]
x2 Hmm.. userextensions value matches actual directory location
![1](https://user-images.githubusercontent.com/30100463/46703500-04e19c80-cc73-11e8-894d-2c608c2ba105.PNG)
![2](https://user-images.githubusercontent.com/30100463/46703771-ce585180-cc73-11e8-9ff5-968caee4d5fb.PNG)

**Comment by eirannejad** [comment]
- Move `extension.json` (assuming you're following the new changes in 4.6) inside the `pyRevitSTR.extension`
- Add the parent path to pyRevit settings only. `H:\pyRevit extensions`. Not the individual extension path

**Comment by antonbondarchuk** [comment]
Revised path as per your comment above and the error is back

![7](https://user-images.githubusercontent.com/30100463/46705031-63aa1480-cc79-11e8-895b-c7de23b0f5c5.PNG)

Thanks for taking time to look into this.. much appreciated!

**Comment by eirannejad** [comment]
Hmmm. Evasive bug. Would you be able to zip and send your extension to my email so I can test locally?

**Comment by antonbondarchuk** [comment]
Well.. it seems I've established what's been causing the issue. I'm very sorry for taking your time, Ehsan. It's been very selfish to assume the issue is with pyRevit itself ;))
My extension is still very much work in progress, and there's not much into it (I'm just learning py and Revit API). I believe the reason for the error was that inside the extension the folder structure was wrong for a splitpushbutton (_layout and folder names were not matching, strange though it was not reporting an error in 4.5 pyRevit). I've fixed it and now it's all good and works even better than before ;) 

Thanks for your outstanding support and prompt response. 
You can close (delete) this issue as it's not relevant to pyRevit whatsoever

**Comment by eirannejad** [comment]
@antonbondarchuk Got your extension. Looks really good and is setup correctly. Loads with no errors on my machine. I tried both network location and local hard drive. I looked at the line that throws the error. The core is trying to list files under a directory but fails on `os.listdir(path)`

The fact that it works perfectly on my machines tell me it’s most probably a path access or path resolution issue. Try placing it on your local drive and test again and let me know if it works.

If it doesn't then go into settings, and activate file logging. Close Revits, cleanup the cache folders (see below) and then load pyRevit and get the `pyRevit*.log` file from the cache folder.

Caches are under:
`%appdata%/pyRevit/<revit-version>`

Delete the folder for your Revit version or use pyrevit cli in command prompt:
`pyrevit caches clear --all`
